### PR TITLE
Show test output for failed tests, remove -v flag

### PR DESCRIPTION
**Pull request by loosebazooka**
Looks like I did the wrong thing in #3548 

Remove the `-v` flags and include detailed failed test output instead.

This only applies to macOS/linux builds, windows runs its tests without using the gotest.sh script: https://github.com/GoogleContainerTools/skaffold/blob/master/.travis.yml#L49 (which probably doesn't need the `-v` flag either, I think I misunderstood what the default was as I didn't realize this script did what it did)

The difference in output is:
OLD:
```
=== Failed Tests ===                                                                                                      
pkg/skaffold/build/jib/TestGetSyncMapFromSystem/direct_only
pkg/skaffold/build/jib/TestGetSyncMapFromSystem
```

NEW:
```
=== Failed Tests ===
/pkg/skaffold/build/jib/TestGetSyncMapFromSystem/direct_only
    --- FAIL: TestGetSyncMapFromSystem/direct_only (0.00s)
        sync_test.go:126: *jib.SyncMap differ (-got, +want):   &jib.SyncMap{                     
                "/tmp/skaffold684430519/dep1": {                                                                          
                        Dest:     []string{"/target/dep1"},                                                               
                        FileTime: s"2020-01-23 15:05:53.741546968 -0500 EST",       
            -           IsDirect: true,                                                                                   
            +           IsDirect: false,                                                                                  
                },                                                                                                        
              }    
/pkg/skaffold/build/jib/TestGetSyncMapFromSystem
    --- FAIL: TestGetSyncMapFromSystem (0.00s)
```

example failure in travis:
- MacOS: https://travis-ci.com/GoogleContainerTools/skaffold/jobs/279436130#L337
- Linux: https://travis-ci.com/GoogleContainerTools/skaffold/jobs/279436129#L324

### CODE REVIEW

File: hack/gotest.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ trap "rm -f $LOG" EXIT
 if [[ " ${@} " =~ "-v" ]]; then
     JQ_FILTER='select(has("Output") and (.Action=="output")) | .Output'
 else
+    #POTATO=1
```
Comment (dgageot) [inline hack/gotest.sh]: ??

File: pkg/skaffold/build/jib/sync_test.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func TestGetSyncMapFromSystem(t *testing.T) {
 				dep1: SyncEntry{
 					[]string{dep1Target},
 					dep1Time,
-					false,
+					true,
```
Comment (dgageot) [inline pkg/skaffold/build/jib/sync_test.go]: Why change that?

File: hack/gotest.sh (hunk 1 above)
Comment (loosebazooka) [reply hack/gotest.sh]: oops

File: pkg/skaffold/build/jib/sync_test.go (hunk 2 above)
Comment (loosebazooka) [reply pkg/skaffold/build/jib/sync_test.go]: just to introduce a failure so I could see if this works, it will be reverted.

**Comment by loosebazooka** [comment]
hold up one second, might be outputting duplicate test outputs...

**Pull request merged**